src/dom/scroll.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global Node */
+
+/**
+ * @module utils/dom/scroll
+ */
+
+import global from './global';
+import isRange from './isrange';
+import Rect from './rect';
+
+/**
+ * Makes any page `HTMLElement` or `Range` (`target`) visible inside the browser viewport.
+ * This helper will scroll all `target` ancestors and the web browser viewport to reveal the target to
+ * the user. If the `target` is already visible, nothing will happen.
+ *
+ * @param {HTMLElement|Range} options.target A target, which supposed to become visible to the user.
+ * @param {Number} [options.viewportOffset] An offset from the edge of the viewport (in pixels)
+ * the `target` will be moved by when the viewport is scrolled. It enhances the user experience
+ * by keeping the `target` some distance from the edge of the viewport and thus making it easier to
+ * read or edit by the user.
+ */
+function scrollViewportToShowTarget( { target, viewportOffset = 0 } ) {
```

Comment:
OK, the problems I remembered were with this notation: `function x( a = { x: 1, y: 2 } )`.